pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/BrokersBase.java

```diff
@@ -437,5 +440,79 @@ private void doShutDownBrokerGracefully(int maxConcurrentUnloadPerSec,
         pulsar().getBrokerService().unloadNamespaceBundlesGracefully(maxConcurrentUnloadPerSec, forcedTerminateTopic);
         pulsar().closeAsync();
     }
+
+    /**
+     * dynamically update log4j2 logger level at runtime.
+     *
+     * @param targetClassName : class name to update
+     * @param targetLevel     : target log level
+     */
+    private CompletableFuture<Void> internalUpdateLoggerLevelAsync(String targetClassName, String targetLevel) {
+        CompletableFuture<Void> loggerLevelFuture = new CompletableFuture<>();
+        CompletableFuture.runAsync(() -> {
+            try {
+                String className;
+                // if set "ROOT" will take effect to rootLogger
+                if (targetClassName.trim().equalsIgnoreCase("ROOT")) {
+                    className = LogManager.ROOT_LOGGER_NAME;
+                } else {
+                    try {
+                        className = targetClassName.trim();
+                        // Check if class name valid
+                        Class.forName(className);
```

Comment:
Here you aree triggering the load of a class from a APi call.
This may be seen as a security issue.
There is no need to do this.
Please remove

Reply:
👍🏻 good suggestions. it is redundant. removed.